pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -171,7 +173,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     bzip2 fontconfig freetype gnutls libiconv lame libass libogg libssh libtheora
-    libvorbis xz soxr x264 x265 xvidcore zlib libopus speex nv-codec-headers
+    libvorbis xz soxr x264 x265 xvidcore zlib libopus speex srt nv-codec-headers
```

Comment:
Shouldn't this be `optional (srt != null) srt`? I'm not sure.

Reply:
The idea would be that it isn't optional, because we're making this default. We could make it possible to override with an optional, but that could also be said of absolutely every single other `buildInput`, hence why every other `buildInput` is not `optional`. Instead, it's better to `overrideAttrs` on the entirety of `buildInputs` to make your changes since there are so many `buildInputs` that equate to removing/adding a feature.

Reply:
I meant that if you build `ffmpeg_3`, then `srt` is not used at all, because this old version of ffmpeg does not have support for it. I'm not sure whether having the dependency in `buildInputs` makes any difference or not, because AFAIK the runtime dependencies should be detected based on hashes in NAR archives.